src/common/RWLock.h

```diff
@@ -42,7 +42,7 @@ class RWLock final
   RWLock(const std::string &n, bool track_lock=true, bool ld=true, bool prioritize_write=false)
     : name(n), id(-1), track(track_lock),
       lockdep(ld) {
-#if defined(PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP)
+#if !defined(__FreeBSD__) && !defined(__APPLE__)
```

Comment:
https://github.com/bminor/glibc/blob/3683b423a4895ecd360fbbe118a027b7c2cf52ec/sysdeps/nptl/pthread.h#L137

could you point me to where the enum is defined or which version of glibc you are looking at?

Reply:
they are not same, defined 
PTHREAD_RWLOCK_ WRITER_NONRECURSIVE_INITIALIZER_NP， 
we use  
PTHREAD_RWLOCK_ PREFER_WRITER_NONRECURSIVE_NP

Reply:
ahh, got it! i am leaning towards checking the glibc specified behavior by checking glibc itself instead of using the heuristic way: assuming that FreeBSD always use its own libc, and Linux always use glibc. actually alpine uses musl which does not implement this GNU extension.

i posted #19202 to address this issue. could you please take a look?